06/ukol13.py

```diff
@@ -0,0 +1,83 @@
+#Napiš funkci piskvorky1d, která vytvoří řetězec s herním polem a střídavě volá funkce tah_hrace a
+#tah_pocitace, dokud někdo nevyhraje nebo nedojde k remíze.
+#Nezapomeň kontrolovat stav hry po každém tahu.
+
+from curses.ascii import isdigit
+from random import randrange
+
+
+def vyhodnot(my_field):
+    '''Vyhodnoti herni pole podle stavu hry.'''    
+    if 'xxx' in my_field:
+        print('Vyhral jsi!')
+        return 'x'
+    elif 'ooo' in my_field:
+        print('Prohral jsi!')
+        return 'o'
+    elif '-' not in my_field:
+        print('Remiza!')
+        return '!'
+    else: 
+        return '-'    
+
+def tah(field, position, symbol):
+    '''Vrati herni pole s danym symbolem umistenym na danou pozici.'''
+    start = field[:position]
+    end = field[position + 1:]
+    if (position >= 0 and position <= 19 and 
+        symbol == 'x' or symbol == 'o'):
+        change = start + symbol + end 
+        return change    
+    else:
+        return False   
+
+def tah_hrace(field):
+    '''Vrati herni pole se zaznamenanym tahem hrace.'''
+    while True:
+        player_position = input('Zadej cislo pozice: ')
+        if player_position.isdigit():
+            break
+        print('Spatne zadana pozice.')
+
+    while True:
+        player_position = int(player_position)    
+        if (player_position >= 0 and player_position <= 19 and 
+           field[player_position] == '-'):
+           return tah(field, player_position, 'x' )    
+        print('Spatne zadana pozice.')
+        player_position = input('Zadej cislo pozice: ')
+
+def tah_pocitace(field):
+    '''Vrati herni pole se zaznamenanym tahem pocitace.'''
+    while True:
+        pc_position = randrange(20)
+        if field[pc_position] == '-':
+            return tah(field, pc_position, 'o')   
+
+def piskvorky1d(): 
+    '''Hra piskvorky 1D'''
+    game_field = (20 * '-')
+    print('0.kolo:', game_field)
+    print('Tvuj symbole je: x')
+    r = 0
+    for i in range(1,21):
```

Comment:
Tady mi není moc jasné, proč je zadaný maximální počet kol.

Na jednu stranu ano - pokud je 20 pozic na plánu a každý tah se zaberou dvě políčka _(člověk + počítač)_, tak tenhle loop určitě nenastane víckrát, než 20x. Na druhou stranu pokud to budeme brát takhle, tak se tenhle loop nestane ani víckrát než 10x, protože právě v každé iteraci nastanou dva tahy.

Na třetí stranu je tam ještě ta nevýhoda, že pokud by ses třeba nabíledni rozhodla herní pole mít větší, tak krom velikosti pole _(a všude těch "magických" konstant 19 či 20 rozesetých po kódu - což by ideálně mohla být někde nahoře v programu "konstanta" definovaná jako `GAME_FIELD_SIZE = 20`, od kterého by se pak vše ostatní odvíjelo - i generování samotného `game_field`, třeba jako `game_field = '-' * GAME_FIELD_SIZE`)_ ... tak k tomu všemu bys ještě musela vědět o tomhle `for i in range(1, 21)` a prostě bys měla další místo, kde by se muselo něco měnit _(což je další místo, kde může vzniknout chyba)_.

Mnohem lepší by bylo celý ten [game-loop](https://en.wikipedia.org/wiki/Video_game_programming#Game_structure) mít koncipovaný tak, že by se to dělo přes `while True: ...` 😉